Sources/IdentifiedCollections/IdentifiedArray/IdentifiedArray+Partial MutableCollection.swift

```diff
@@ -118,19 +118,44 @@ extension IdentifiedArray {
   /// - Complexity: O(*n* log *n*), where *n* is the length of the collection.
   @inlinable
   public mutating func move(fromOffsets source: IndexSet, toOffset destination: Int) {
-    var removed: [Element] = []
-    var removedBeforeDestinationCount = 0
+    var destination = destination
+    let suffixStart = self.halfStablePartition { index, _ in
+      if index < destination { destination -= 1 }
```

Comment:
This is the main change from the original code, in order to support the behavior we see in SwiftUI we need to nudge the insertion point.

The original code looks largely the same to the this code committed to Swift awhile back: https://github.com/apple/swift/commit/0c7a091390057b433f253937d960ec9d1bb35304